fe/fe-core/src/main/java/com/starrocks/sql/optimizer/rule/transformation/MergeLimitWithSortRule.java

```diff
@@ -30,8 +30,9 @@ public boolean check(final OptExpression input, OptimizerContext context) {
         LogicalLimitOperator limit = (LogicalLimitOperator) input.getOp();
         LogicalTopNOperator sort = (LogicalTopNOperator) input.getInputs().get(0).getOp();
 
-        OptExpression result = new OptExpression(
-                new LogicalTopNOperator(sort.getOrderByElements(), limit.getLimit(), limit.getOffset()));
+        long l = limit.hasLimit() ? limit.getLimit() : -1;
+        long of = limit.hasOffset() ? limit.getOffset() : 0;
```

Comment:
What't meaning of these two line codes?  limit default value is -1, offset default value is 0.

Reply:
limit default offset is -1, TOPN handle -1 offset is error

Reply:
So, what's the meaning of `long l = limit.hasLimit() ? limit.getLimit() : -1;`?

The default offset for limit why not 0?

Reply:
Limit 0 is really can be appear in sql, like select * from xx limit 0, it's right and result return 0 rows

Reply:
BTW, I think we can change offset default 0, and support static `NONE_LIMIT` and `NONE_OFFSET`

Reply:
I will modify it in next PR